mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown

```diff
@@ -149,16 +149,21 @@ group. You can specify only one value. Structure is [documented below](#nested_a
 
 ```hcl
 update_policy {
-  type                  = "PROACTIVE"
-  minimal_action        = "REPLACE"
-  max_surge_percent     = 20
-  max_unavailable_fixed = 2
-  min_ready_sec         = 50
-  replacement_method    = "RECREATE"
+  type                           = "PROACTIVE"
+  minimal_action                 = "REPLACE"
+  most_disruptive_allowed_action = "REPLACE"
+  max_surge_percent              = 20
+  max_unavailable_fixed          = 2
+  <% unless version == "ga" -%>
+  min_ready_sec                  = 50
+  <% end -%>  
+  replacement_method             = "RECREATE"
 }
 ```
 
-* `minimal_action` - (Required) - Minimal action to be taken on an instance. You can specify either `RESTART` to restart existing instances or `REPLACE` to delete and create new instances from the target template. If you specify a `RESTART`, the Updater will attempt to perform that action only. However, if the Updater determines that the minimal action you specify is not enough to perform the update, it might perform a more disruptive action.
+* `minimal_action` - (Required) - Minimal action to be taken on an instance. You can specify either `REFRESH` to update without stopping instances, `RESTART` to restart existing instances or `REPLACE` to delete and create new instances from the target template. If you specify a `REFRESH`, the Updater will attempt to perform that action only. However, if the Updater determines that the minimal action you specify is not enough to perform the update, it might perform a more disruptive action.
+
+* `most_disruptive_allowed_action` - (Required) - Most disruptive action that is allowed to be taken on an instance. You can specify either NONE to forbid any actions, REFRESH to allow actions that do not need instance restart, RESTART to allow actions that can be applied without instance replacing or REPLACE to allow all possible actions. If the Updater determines that the minimal update action needed is more disruptive than most disruptive allowed action you specify it will not perform the update at all.
```

Comment:
This field is not required, so the docs shouldn't state it is required.

What is the behavior if the user doesn't specify this field?

Reply:
I changed documentation so that now it specifies this field as Optional. Documentation text is taken from public [REST api docs ](https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers#:~:text=enum-,Most%20disruptive%20action%20that,updatePolicy.maxSurge,-object)

If most_disruptive_allowed_action is not specified then it is unbound - there is no limit for maximal action that can be performed. No explicit value will be set in UpdatePolicy on the MIG. Updater will behave in the same way as if REPLACE is specified.